### PR TITLE
Artifactory Release Lifecycle Management - get lifecycle release bundle specification

### PR DESCRIPTION
**Pull request by ns-kbhat**
- [ ] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Comment by ns-kbhat** [comment]
@RobiNino please review.

File: lifecycle/manager.go [hunk 1]
```diff
@@ -54,6 +54,12 @@ func (lcs *LifecycleServicesManager) CreateReleaseBundleFromBundles(rbDetails li
 	return rbService.CreateFromBundles(rbDetails, queryParams, signingKeyName, sourceReleaseBundles)
 }
 
+func (lcs *LifecycleServicesManager) GetReleaseBundleSpecification(rbDetails lifecycle.ReleaseBundleDetails) (lifecycle.ReleaseBundleSpecResponse, error) {
+	rbService := lifecycle.NewReleaseBundlesService(lcs.config.GetServiceDetails(), lcs.client)
+	bs, _, err := rbService.GetReleaseBundleSpec(rbDetails)
+	return bs, err
```
Comment (RobiNino) [inline lifecycle/manager.go:60]:
```suggestion
	return rbService.GetReleaseBundleSpec(rbDetails)
```
I suggest to remove the unused `body` output from the function if it isn't used.

File: lifecycle/services/status.go [hunk 2]
```diff
@@ -179,6 +198,23 @@ type ReleaseBundleStatusResponse struct {
...
+	CreatedMillis int       `json:"created_millis,omitempty"`
+	Artifacts     []struct {
+		Path                string `json:"path,omitempty"`
+		Checksum            string `json:"checksum,omitempty"`
+		SourceRepositoryKey string `json:"source_repository_key,omitempty"`
+		PackageType         string `json:"package_type,omitempty"`
+		Size                int    `json:"size,omitempty"`
+		Properties          []struct {
+			Key   string `json:"key"`
+			Value string `json:"value"`
```
Comment (RobiNino) [inline lifecycle/services/status.go:213]:
```suggestion
			Values []string `json:"values"`
```

**Review by RobiNino — CHANGES_REQUESTED** [review]
Thank you for another contribution @ns-kbhat !
Left a few minor comments on the code.

Since there is a lot of room for error in the parsing of this body, I suggest we add a unit test with a mock server in `lifecycle_test.go`. You can use the sample response from the [documentation](https://jfrog.com/help/r/jfrog-rest-apis/get-release-bundle-v2-version-specification).
Let me know if my assistance is required, thanks!

**Comment by ns-kbhat** [comment]
> Thank you for another contribution @ns-kbhat ! Left a few minor comments on the code.
> 
> Since there is a lot of room for error in the parsing of this body, I suggest we add a unit test with a mock server in `lifecycle_test.go`. You can use the sample response from the [documentation](https://jfrog.com/help/r/jfrog-rest-apis/get-release-bundle-v2-version-specification). Let me know if my assistance is required, thanks!

@RobiNino added a test.
I don't use other source types yet, so I left them out of the spec type and checks.

**Comment by ns-kbhat** [comment]
@RobiNino happened to notice an error in the `dev` branch.
Looks like signature change of `GetFileChecksums()` from `github.com/jfrog/build-info-go/utils` is breaking existing calls, please check or notify relevant owner.

**Review by RobiNino — APPROVED** [review]
Thank you @ns-kbhat !

**Comment by ns-kbhat** [comment]
@RobiNino I keep getting failed workflow notification on this PR.

Is something pending that needs to be taken care of to close the PR correctly?

**Comment by RobiNino** [comment]
@ns-kbhat For some reason every commit triggered a deployment that failed. 
Anyway, looks like the last one ended so you shouldn't receive any other notifications.
Sorry for the interruption!

**Pull request merged**